server/src/main/java/com/decathlon/ara/cartography/SquashExporter.java

```diff
@@ -27,8 +54,117 @@ public String getFormat() {
     }
 
     @Override
-    public byte[] generate(List<FunctionalityDTO> functionalities) {
-        // TODO
+    public byte[] generate(List<FunctionalityDTO> functionalities, Map<String, String> requiredInfos) {
+        String projectNameInSquash = requiredInfos.get(PROJECT_NAME);
+        String userWhichMakesTheImport = requiredInfos.get(USER);
+
+        HSSFWorkbook workbook = new HSSFWorkbook();
+        Sheet sheet = workbook.createSheet("REQUIREMENT");
+
+        this.createHeaderRow(sheet);
+        Date importDate = new Date();
+        int idx = 1;
+        for (FunctionalityDTO functionality: functionalities) {
+            if (FunctionalityType.FUNCTIONALITY.name().equals(functionality.getType())) {
+                String criticity = CRITICALITY_MAPPING.get(functionality.getSeverity());
+                String funcName = functionality.getName();
+                String path = "/" + projectNameInSquash + this.getFunctionalityPath(functionality, functionalities);
```

Comment:
String.format ?

Reply:
Despite the fact that String.format doesn't exists in Java (StringBuilder ...), we're effectively in a loop here so yep, totally agree with you 👍

Reply:
Done